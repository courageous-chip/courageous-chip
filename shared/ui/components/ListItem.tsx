import React, { FC } from "react";
import { StyleSheet, Text, TouchableHighlight } from "react-native";

type Props = { id: string; onPress: (id: string) => void; text: string };

export const ListItem: FC<Props> = (props) => {
  const { id, text } = props;
  const onPress = () => props.onPress(id);

  return (
    <TouchableHighlight
      onPress={onPress}
      style={styles.container}
      underlayColor={styles.underlay.backgroundColor}
    >
      <Text>{text}</Text>
    </TouchableHighlight>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: "flex-start",
    flex: 1,
    justifyContent: "center",
    minHeight: 45,
    paddingHorizontal: 20,
    paddingVertical: 10,
  },
  underlay: {
    backgroundColor: "#ddd",
  },
});
