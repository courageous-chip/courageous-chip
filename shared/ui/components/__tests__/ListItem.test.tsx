import React from "react";
import renderer from "react-test-renderer";

import { ListItem } from "../ListItem";

describe("<ListItem />", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<ListItem id="1" onPress={() => {}} text="One" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
