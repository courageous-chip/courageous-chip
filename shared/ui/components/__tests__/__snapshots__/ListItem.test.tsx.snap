// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "flex-start",
      "flex": 1,
      "justifyContent": "center",
      "minHeight": 45,
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={null}
  >
    One
  </Text>
</View>
`;
